### ISSUE TITLE
Missing Solutions

### ISSUE THREAD

**Issue by DanRobinsonHPE** [apnex/vmw-cli#8]
I read your feedback on Issue #2 , but it doesn't seem to work for me.

I'm looking for vCloud Director 9.7.0.4 for example.

When I go to the web page, I can find this URL for the files I want:
https://my.vmware.com/web/vmware/details?downloadGroup=VSPP_VCD9704&productId=867&rPId=41588

So I run:
vmw-cli list
vmw-cli index vmware-vcloud-director

But there is no VSPP_VCD9704 among the results, closest is VSPP_VCD9703

So based on your answer to #2 , I run:
vmw-cli index VSPP_VCD9704
Solution [VSPP_VCD9704] not found! Please re-index...

If that's not the Product ID, perhaps you can elaborate on where to find the proper Product ID from the web page.

OR
If there is simply a way to hand vmw-cli the URL I started with above, or the product ID 867 for example, and say Index THAT and then show me the files, I'm open to that too.

**Comment by DanRobinsonHPE** [comment]
Thanks for the previous update.
I thought I posted back it was working but I guess not.

I just found Mirage has a similar issue, doesn't seem to have been updated since 5.4

https://my.vmware.com/group/vmware/details?downloadGroup=MIRAGE-591&productId=955&rPId=43191

```
vmw-cli index MIRAGE-591
Product group already indexed, skipping... [MIRAGE-591]

vmw-cli find fileName:VMware.Mirage.Core*
solution        productGroup  productType       version  fileName                          fileDate    fileSize   fileType  download
--------------  ------------  ----------------  -------  --------------------------------  ----------  ---------  --------  --------
vmware-horizon  MIRAGE-540    Product Binaries  5.4.0    VMware.Mirage.Core.540.44732.zip  2015-06-23  268.03 MB  zip       yes
vmware-horizon  MIRAGE-530    Product Binaries  5.3.0    VMware.Mirage.Core.530.31896.zip  2015-03-03  187 MB     zip       yes
[ 2/17355 ]
```
I get the same thing for the other 2 components in that Product Group.

**Comment by DanRobinsonHPE** [comment]
vSphere 7 Suite seems to also be missing.

Solution [VC700] not found! Please re-index

[ 0/21630 ] entries - filter [ fileName:VMware-VMvisor-Installer-7 ]

I know it only launched today.
Wondering if there isn't an automated way for you to update the mainIndex

**Comment by apnex** [comment]
Hi Dan,

I have performed a full refresh of the mainIndex.json file - which should include these (and vSphere/vCenter 7.0 binaries now).

I do note that the VMware website has some oddities with productGroup IDs for these, so I had to put in some workarounds to allow for permissions to work properly - this means I recommend removing and reinstalling vmw-cli (1.2.16) from npm if you went that route.

I do indeed run automate this scan by running the 'get-solution.js' script in the node /lib directory (which you could run yourself), but I haven't added this is a command option in the client yet - I'm in the process of reworking some of these workflows. For now - I'm hosting the mainIndex.json file on github for the client to pull down when needed.

Immediately priority is to rewrite the authentication and file downloading logic as both `tough-cookie` and `requests` modules have become deprecated and have issues.

Goal is to have a hybrid indexing solution that hosts an API with an always-up-to-date mainIndex, with clients handling their own authenticating specific file calls.

Let me know if there is any specific file missing now

**Comment by DanRobinsonHPE** [comment]
I tried to uninstall, but it left some stuff behind, so I just nuked the folder:
/usr/local/lib/node_modules/vmw-cli

Then I installed again using the command: sudo npm install vmw-cli --global
It says towards the end of the install:
+ vmw-cli@1.2.14

So I'm not sure if I needed to do something else to get 1.2.16 or if something just didn't get updated properly to reflect that.

I then did my exports and tried to find NSX-T 3.0
NSX-T-300 productId=982
No luck.

Not sure if I didn't get the new version and that is part of the problem or if it's simply missing.
Going to try and add it to my mainIndex manually and see if it works.
